tests/test_web_log.py

```diff
@@ -43,13 +43,44 @@ def test_access_logger_format() -> None:
     Ref: https://stackoverflow.com/a/46102240/595220
     """,  # noqa: E501
 )
-def test_access_logger_atoms(mocker) -> None:
-    utcnow = datetime.datetime(1843, 1, 1, 0, 30)
-    mock_datetime = mocker.patch("aiohttp.datetime.datetime")
-    mock_getpid = mocker.patch("os.getpid")
-    mock_datetime.utcnow.return_value = utcnow
-    mock_getpid.return_value = 42
-    log_format = '%a %t %P %r %s %b %T %Tf %D "%{H1}i" "%{H2}i"'
+@pytest.mark.parametrize(
+    'log_format,expected,extra',
+    [
+        ('%t',
+         '[01/Jan/1843:00:29:56 +0800]',
+         {
+             'request_start_time': '[01/Jan/1843:00:29:56 +0800]'
+         }
+         ),
+        ('%a %t %P %r %s %b %T %Tf %D "%{H1}i" "%{H2}i"',
+         ('127.0.0.2 [01/Jan/1843:00:29:56 +0000] <42> '
+          'GET /path HTTP/1.1 200 42 3 3.141593 3141593 "a" "b"'),
+         {
+             'first_request_line': 'GET /path HTTP/1.1',
+             'process_id': '<42>',
+             'remote_address': '127.0.0.2',
+             'request_start_time': '[01/Jan/1843:00:29:56 +0000]',
+             'request_time': 3,
+             'request_time_frac': '3.141593',
+             'request_time_micro': 3141593,
+             'response_size': 42,
+             'response_status': 200,
+             'request_header': {'H1': 'a', 'H2': 'b'},
+         }
+         )
+    ]
+)
+def test_access_logger_atoms(monkeypatch, log_format, expected, extra) -> None:
+    class PatchedDatetime(datetime.datetime):
+        @staticmethod
+        def now(tz):
+            return datetime.datetime(1843, 1, 1, 0, 30, tzinfo=tz)
+
+    def new_get_pid():
+        return 42
+    monkeypatch.setattr('datetime.datetime', PatchedDatetime)
+    monkeypatch.setattr('time.timezone', -28800)
+    monkeypatch.setattr("os.getpid", new_get_pid)
```

Comment:
Don't mix up the quotation style. Also, this can be simply
```suggestion
    monkeypatch.setattr('os.getpid', lambda: 42)
```